feature/google/src/bc5SupportDev/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -140,14 +141,62 @@ class BillingWrapper(
             }
         }
     }
+    private fun getIsBC5Enabled(offeringsJSON: JSONObject): Boolean {
+        val jsonArrayOfOfferings = offeringsJSON.getJSONArray("offerings")
+        for (i in 0 until jsonArrayOfOfferings.length()) {
+            val jsonPackagesArray =
+                jsonArrayOfOfferings.getJSONObject(i).getJSONArray("packages")
+            for (j in 0 until jsonPackagesArray.length()) {
+                val pkg = jsonPackagesArray.getJSONObject(j)
+                if (pkg.has(SUBSCRIPTION_ID_BACKEND_KEY))
+                    return true
+            }
+        }
+        return false
+    }
+
+    private fun extractSubscriptionIds(offeringsJSON: JSONObject): Set<String> {
+        val jsonArrayOfOfferings = offeringsJSON.getJSONArray("offerings")
```

Comment:
i wonder if we could re-use some of this parsing code somehow, since isBC5Enabled will always be followed by subscription ID extraction 🤔

Reply:
yep, definitely! if we actually also parse offerings sooner, this all becomes a lot easier